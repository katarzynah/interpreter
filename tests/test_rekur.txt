program f;
var x: int;

procedure rekur(x: int;);
begin
if x > 0 then
begin
    print(x);
    rekur(x - 1,);
end
else
    print(x)
endif;
end;

begin
    x := 6;
    rekur(x,);
end.