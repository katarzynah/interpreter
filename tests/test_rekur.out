[(fromList [(Ident "x",undefined)],fromList [(Ident "rekur",ProcDecProc (ProcHead (Ident "rekur") (Args (ArgList (ArgLabel (IdLEnd (Ident "x")) TypeSpecInt) ArgListEmpty))) VarDecEmpty (CompStmnt (StmntList (SIf (IfStmntWithElse (ExpGreater (SimpleExpTerm (TermFactor (FactorIdent (Ident "x")))) (SimpleExpTerm (TermFactor (FactorConstant (ConstInt 0))))) (SComp (CompStmnt (StmntList (SPrint (PrintStmnt (ExpSimple (SimpleExpTerm (TermFactor (FactorExpression (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "x"))))))))))) (StmntList (SProc (ProcCall (Ident "rekur") (Act (ExpList (ExpSimple (SimpleExpSubst (SimpleExpTerm (TermFactor (FactorIdent (Ident "x")))) (TermFactor (FactorConstant (ConstInt 1))))) ExpListEmpty)))) StmntListEmpty)))) (SPrint (PrintStmnt (ExpSimple (SimpleExpTerm (TermFactor (FactorExpression (ExpSimple (SimpleExpTerm (TermFactor (FactorIdent (Ident "x"))))))))))))) StmntListEmpty)))])]
6
5
4
3
2
1
0
